schematic/schemas/data_model_parser.py

```diff
@@ -0,0 +1,319 @@
+        # Determine base schema path
+        base_model_path = self._get_base_schema_path(self.base_schema_path)
+
+        # Parse
+        jsonld_parser = DataModelJSONLDParser()
+        base_model = jsonld_parser.parse_jsonld_model(base_model_path)
+        return base_model
+
+    def parse_model(self):
+        '''Given a data model type, instantiate and call the appropriate data model parser.
+        Returns:
+            model_dict, dict:
+                {Attribute Display Name: {
+                        Relationships: {
+                                    CSV Header: Value}}}
+        Note: in future will add base model parsing in this step too and extend new model off base model.
+        '''
+        #base_model = self.parse_base_model()
+
+        # Call appropriate data model parser and return parsed model.
+        if self.model_type == 'CSV':
+            csv_parser = DataModelCSVParser()
+            model_dict = csv_parser.parse_csv_model(self.path_to_data_model)
+        elif self.model_type == 'JSONLD':
+            jsonld_parser = DataModelJSONLDParser()
+            model_dict = jsonld_parser.parse_jsonld_model(self.path_to_data_model)
+        else:
+            raise ValueError(f"Schematic only accepts models of type CSV or JSONLD, you provided a model type {self.model_type}, please resubmit in the proper format.")
+
+        return model_dict
+    
+class DataModelCSVParser():
+    def __init__(
+        self
+        ):
+        # Instantiate DataModelRelationships
+        self.dmr = DataModelRelationships()
+        # Load relationships dictionary.
+        self.rel_dict = self.dmr.define_data_model_relationships()
+        # Get edge relationships
+        self.edge_relationships_dictionary = self.dmr.define_edge_relationships()
+        # Load required csv headers
+        self.required_headers = self.dmr.define_required_csv_headers()
+
+    def check_schema_definition(self, model_df: pd.DataFrame) -> bool:
+        """Checks if a schema definition data frame contains the right required headers.
+        Args:
+            schema_definition: a pandas dataframe containing schema definition; see example here: https://docs.google.com/spreadsheets/d/1J2brhqO4kpeHIkNytzlqrdIiRanXDr6KD2hqjOTC9hs/edit#gid=0
+        Raises: Exception
+        """
+        try:
+            if set(self.required_headers).issubset(set(list(model_df.columns))):
+                return
+            elif "Requires" in list(model_df.columns) or "Requires Component" in list(
+                model_df.columns
+            ):
+                raise ValueError(
+                    "The input CSV schema file contains the 'Requires' and/or the 'Requires "
+                    "Component' column headers. These columns were renamed to 'DependsOn' and "
+                    "'DependsOn Component', respectively. Switch to the new column names."
+                )
+            logger.debug("Schema definition csv ready for processing!")
+        except:
```

Comment:
This looks like it should be expecting a specific error. You generally don't want to except all errors.

Reply:
I was able to get rid of the try with an additional elif.